app/viewmodels/checkAnswers/income/HowMuchTradingAllowanceSummary.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers.income
+
+import controllers.journeys.income.routes.HowMuchTradingAllowanceController
+import models.{CheckMode, HowMuchTradingAllowance, UserAnswers}
+import pages.income.{HowMuchTradingAllowancePage, TurnoverIncomeAmountPage}
+import play.api.i18n.Messages
+import uk.gov.hmrc.govukfrontend.views.Aliases.{Key, Value}
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryListRow
+import viewmodels.govuk.summarylist._
+import viewmodels.implicits._
+
+object HowMuchTradingAllowanceSummary {
+
+  def row(answers: UserAnswers, taxYear: Int, authUserType: String, businessId: String)(implicit messages: Messages): Option[SummaryListRow] =
+    answers.get(HowMuchTradingAllowancePage, Some(businessId)).map { answer =>
+      val value = answer match {
+        case HowMuchTradingAllowance.Maximum =>
+          val turnoverIncomeAmount = answers.get(TurnoverIncomeAmountPage, Some(businessId)) match {
+            case Some(amount) => amount.setScale(2)
+            case None         => throw new RuntimeException("Unable to retrieve user answers for TurnoverIncomeAmountPage")
```

Comment:
Minor: it would be more functional (easier to test too) to return Either `Left` than throwing exception. Ignore if it requires a lot of changes.

Reply:
Where would we handle the left case, and what would we do?

Reply:
We could redirect to the JourneyRecoveryController?

Reply:
Ideally we pass Left through the application until controller. 

To not change much and keep the job easy,we can throw this exception there in conroller (if left => runtime exception). Not sure if there's better way to trigger our common error handler.

Reply:
I have pushed up a suggestion to this, let me know if all OK.